### ISSUE TITLE
Is the extra class embedding important to predict the results, why not simply use feature maps to predict?

### ISSUE THREAD

**Issue by QiushiYang** [google-research/vision_transformer#61]
Different from the common ways to use feature maps to obtain classifcation prediction (with fc or GAP layers), VIT employs an extra class embedding to do this without using feature maps explicitly. Wonder the meanings of this unusual design?

BTW, I used official pre-training params to fine-tune VIT on a small dataset, found that the validation accuracy is a little better after I replaced the feature maps with leanable class embedding to predict. So is the class embedding (maybe like a kind of query within encoder) important to learn and to predict?

**Comment by lucasb-eyer** [comment]
Great question. It is not really important. However, we wanted the model to be "exactly Transformer, but on image patches", so we kept this design from Transformer, where a <cls> token is always used.

**Comment by moha23** [comment]
Hi @lucasb-eyer, a possibly naive question, I'm trying to wrap my head around the use of 'cls' token. From the code https://github.com/google-research/vision_transformer/blob/4317e064a0a54b825b5b9ff634482954788b8d84/vit_jax/models.py#L251, I understand that if classification token is to be used, first a read-only variable 'cls' of shape (1, 1, c) is defined, initialised with zeros and repeated along first axis to get shape (n,1,c) and finally concatenated to patch embeddings x. So, the values of 'cls' token keep getting updated as training proceeds? Or is it fixed to be 0 valued? When you say 'learnable' embedding, does it mean it is a trainable variable and the network learns in the training process what its values should be?

**Comment by lucasb-eyer** [comment]
1-3 are all the same question :) yes, it is zero-initialized, then learned/updated. It does not depend on class at the input side, it's purpose is to "collect" evidence for what the output should be while going through the network.
